### ISSUE TITLE
Instantiating ShoeBox with two methods differ

### ISSUE THREAD

**Issue by Owen864720655** [LCAV/pyroomacoustics#69]
Hi authors,

Given the parameters:
```python
room_dim = [15, 14, 16]
absorption = 0.2
source_position = [2.0, 3.1, 2.0]
mic_position = [2.0, 1.5, 2.0]
fs = 16000
max_order = 15
```

It seems that the two instances go different ways:

Scenario A
```python
source = SoundSource(position=source_position)
mics = MicrophoneArray(np.array([mic_position]).T, fs)
shoebox = ShoeBox(room_dim, absorption=absorption, fs=fs, max_order=max_order, sources=[source], mics=mics)
```

Scenario B
```python
shoebox = ShoeBox(room_dim, absorption=absorption, fs=fs, max_order=max_order)
shoebox.add_source(source_position)
mics = MicrophoneArray(np.array([mic_position]).T, fs)
shoebox.add_microphone_array(mics)
```

Then run and plot the image source model as
```python
shoebox.image_source_mode()
shoebox.plot(img_order=2)
```

My experiments show that the instantiation of using `SoundSource` as the arguments fails, and only the method `add_source` is feasible. Please fix the bugs if exist

**Comment by fakufaku** [comment]
Interesting. Thanks for reporting this. I will look into it.

**Comment by fakufaku** [comment]
Fixed by PR #71
